sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/throughputControl/controller/group/global/ThroughputControlContainerManager.java

```diff
@@ -139,15 +140,21 @@ public ThroughputControlContainerManager(GlobalThroughputControlGroup group) {
     public Mono<Double> queryLoadFactorsOfAllClients(double clientLoadFactor) {
         // The current design is using ttl to expire client items, so there is no need to check whether the client item is expired.
 
-        String sqlQueryTest = "SELECT VALUE SUM(c.loadFactor) FROM c WHERE c.groupId = @GROUPID AND c.id != @CLIENTITEMID";
+        // You probably have the question about why not use the following query:
+        // "SELECT VALUE SUM(c.loadFactor) FROM c WHERE c.groupId = @GROUPID AND c.id != @CLIENTITEMID"
+        // The reason being that we might get some inconsistent results back for the query above, using the following query will make sure
+        // we always get the items within ttl.
```

Comment:
Ouch - didn't know that query isn't always honoring TTL in aggregates.

Reply:
I think it might due to the cleaning of indexing is async process, so sometimes you will get items back even when they have soft-expired.